src/object/obj_enum.c

```diff
@@ -535,16 +535,91 @@ csum_copy_inline(int type, vos_iter_entry_t *ent, struct dss_enum_arg *arg,
 	return 0;
 }
 
+static bool
+need_new_entry(struct dss_enum_arg *arg, vos_iter_entry_t *key_ent,
+	       daos_size_t iod_size, int type)
+{
+	struct obj_enum_rec	*rec;
+	d_iov_t			*iovs = arg->sgl->sg_iovs;
+	uint64_t		curr_off = key_ent->ie_recx.rx_idx;
+	uint64_t		curr_size = key_ent->ie_recx.rx_nr;
+	uint64_t		priv_off;
+	uint64_t		priv_size;
+
+	if (arg->last_type != OBJ_ITER_RECX || type != OBJ_ITER_RECX)
+		return true;
+
+	rec = iovs[arg->sgl_idx].iov_buf + iovs[arg->sgl_idx].iov_len - sizeof(*rec);
+	priv_off = rec->rec_recx.rx_idx;
+	priv_size = rec->rec_recx.rx_nr;
+	if (priv_off + priv_size != curr_off) /* not continuous */
+		return true;
+
+	if (arg->rsize != iod_size)
+		return true;
+
+	if (arg->ec_cell_sz > 0 &&
+	    (priv_off + priv_size - 1) / arg->ec_cell_sz !=
+	    (curr_off + curr_size) / arg->ec_cell_sz)
```

Comment:
you mean if they are on different stripe (for example cell_sz 1M, [1M to 2M] and [2M to 3M]), they cannot merge? could you please remind me why? thanks

Reply:
yes, we can not merge recx cross the cell size boundary,  because we need the EPOCH of each individual cell for EC rebuild/reintegration, otherwise it will cause data corruption.

Reply:
ok, not sure why above with "prev_off + prev_size - 1" and "curr_off + curr_size" (with and without -1).
for example cell_sz 1M, prev_off 1M, prev_size 512K; curr_off 1M+512K, curr_size 512K.
in this case the two recx should be able to merge right? but the code will return false?

another case is prev [1M to 10M+1K], curr is [10M+1k, 10M+2k], can it be merged? (the prev recx itself cross cell sz).